app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -330,6 +334,13 @@ class SearchFragment : Fragment(), UserInteractionHandler {
             )
         }
 
+        // Users can from this fragment go to install/uninstall search engines and then return.
+        with(areShortcutsAvailable()) {
+            if (searchStore.state.areShortcutsAvailable != this) {
+                searchStore.dispatch(SearchFragmentAction.UpdateShortcutsAvailability(this))
```

Comment:
Perhaps this logic should be handled in the reducer? We could always dispatch, but only return a new state if the `areShortcutsAvailable` value changes.

Reply:
Thanks!
Thinking about it, the reducer has to return a state. It could do that check and not modify anything in the state, but it has to return a state.
State which would trigger an update for the view which could be unnecessary.
By only dispatching when we want to modify the state we can control and limit view updates.

Reply:
You're right, I thought `consumeFrom` checked if the state had changed first.